In [3]:
import pandas as pd
data = pd.read_csv('vgsales.csv')
data.head()

Rank                      Name Platform    Year         Genre Publisher  \
0     1                Wii Sports      Wii  2006.0        Sports  Nintendo   
1     2         Super Mario Bros.      NES  1985.0      Platform  Nintendo   
2     3            Mario Kart Wii      Wii  2008.0        Racing  Nintendo   
3     4         Wii Sports Resort      Wii  2009.0        Sports  Nintendo   
4     5  Pokemon Red/Pokemon Blue       GB  1996.0  Role-Playing  Nintendo   

   NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
0     41.49     29.02      3.77         8.46         82.74  
1     29.08      3.58      6.81         0.77         40.24  
2     15.85     12.88      3.79         3.31         35.82  
3     15.75     11.01      3.28         2.96         33.00  
4     11.27      8.89     10.22         1.00         31.37

In [5]:
data.shape #у нас более 100 значений

(16598, 11)

# 1. Фильтрация.

In [6]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16598 entries, 0 to 16597
Data columns (total 11 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Rank          16598 non-null  int64  
 1   Name          16598 non-null  object 
 2   Platform      16598 non-null  object 
 3   Year          16327 non-null  float64
 4   Genre         16598 non-null  object 
 5   Publisher     16540 non-null  object 
 6   NA_Sales      16598 non-null  float64
 7   EU_Sales      16598 non-null  float64
 8   JP_Sales      16598 non-null  float64
 9   Other_Sales   16598 non-null  float64
 10  Global_Sales  16598 non-null  float64
dtypes: float64(6), int64(1), object(4)
memory usage: 1.4+ MB


Артемий, сделай мне фильтрацию и описание признаков, я не понимаю, что от меня хотят
#хэлп

Рассмотрим те видеоигры, которые были выпущены на платформе PS2, не раньше 2000 года и имеют жанр игр Action.

In [25]:
data_one=data[(data.Platform=='PS2')&(data.Year>=2000.0)&(data.Genre=='Action')]
data_new.head()

Rank                                   Name Platform    Year   Genre  \
17     18          Grand Theft Auto: San Andreas      PS2  2004.0  Action   
24     25            Grand Theft Auto: Vice City      PS2  2002.0  Action   
38     39                   Grand Theft Auto III      PS2  2001.0  Action   
142   143    Metal Gear Solid 2: Sons of Liberty      PS2  2001.0  Action   
227   228  The Lord of the Rings: The Two Towers      PS2  2002.0  Action   

                        Publisher  NA_Sales  EU_Sales  JP_Sales  Other_Sales  \
17           Take-Two Interactive      9.43      0.40      0.41        10.57   
24           Take-Two Interactive      8.41      5.49      0.47         1.78   
38           Take-Two Interactive      6.99      4.51      0.30         1.30   
142  Konami Digital Entertainment      2.45      2.01      0.87         0.72   
227               Electronic Arts      1.94      1.95      0.08         0.70   

     Global_Sales  
17          20.81  
24          16.15  
38          13.10  
142          6.05  
227          4.67

In [26]:
data_one['Publisher'].value_counts()

Capcom                                    29
Activision                                28
THQ                                       26
Sony Computer Entertainment               22
Take-Two Interactive                      21
Namco Bandai Games                        21
Tecmo Koei                                20
Konami Digital Entertainment              20
Electronic Arts                           19
Eidos Interactive                         17
Sega                                      14
Ubisoft                                   13
Vivendi Games                              9
Atari                                      9
Midway Games                               8
LucasArts                                  6
Disney Interactive Studios                 6
D3Publisher                                6
Warner Bros. Interactive Entertainment     5
505 Games                                  4
Takara                                     3
Universal Interactive                      3
Acclaim En

В выборке из опроса есть три производителя, чьи игры представлены в максимальном количестве в данной выборке. Это Capcom, Activision и THQ. По этому сократим нашу выборку до игр, которые представлены этими тремя производителями.

In [36]:
data_new = data_one.loc[data_one['Publisher'].isin(["Capcom","Activision","THQ"])]
data_new.head()

Rank                       Name Platform    Year   Genre   Publisher  \
241   242      Spider-Man: The Movie      PS2  2002.0  Action  Activision   
350   351            Resident Evil 4      PS2  2005.0  Action      Capcom   
390   391               Spider-Man 2      PS2  2004.0  Action  Activision   
411   412  True Crime: Streets of LA      PS2  2003.0  Action  Activision   
475   476              Devil May Cry      PS2  2001.0  Action      Capcom   

     NA_Sales  EU_Sales  JP_Sales  Other_Sales  Global_Sales  
241      2.71      1.51      0.03         0.23          4.48  
350      2.08      0.83      0.46         0.25          3.62  
390      1.75      1.20      0.02         0.43          3.41  
411      1.89      1.05      0.02         0.31          3.27  
475      1.36      0.86      0.64         0.13          2.99

# 2. Статистика.



Давайте узнаем средний балл по продажам каждой из компании в каждом регионе продаж.

In [37]:
data_new[['NA_Sales','EU_Sales','JP_Sales','Publisher']].groupby('Publisher').mean()

NA_Sales  EU_Sales  JP_Sales
Publisher                               
Activision  0.447500  0.217500  0.013929
Capcom      0.339655  0.223793  0.176897
THQ         0.300385  0.173846  0.057692

В выборке из опроса самые большие продажи по Северной Америке у производителя "Activision", по Европе у производителя "Capcom" и по Японии у производителя "Capcom". Самые низкие продажи по Северной Америке у производителя "THQ", по Европе у производителя "THQ" и по Японии у производителя "Activision".

Теперь мы можем посмотреть, у какого производителя в общем наилучшие продажи по миру и как они зависят от продаж в отдельных регионах.

In [39]:
data_new[['Global_Sales','Other_Sales','Publisher']].groupby('Publisher').mean()

Global_Sales  Other_Sales
Publisher                            
Activision      0.797857     0.117500
Capcom          0.814483     0.074138
THQ             0.623846     0.093077

Больше всего продаж по миру у производителя "Activision", в то время как по продажам в других регионах они занимают второе место, уступая первое место производителю "Capcom".

# 3. Корреляция.



Посмотрим на корреляцию между продажами в 3 странах в нашей выборке.

NA_Sales - Sales in North America (in millions)

EU_Sales - Sales in Europe (in millions)

JP_Sales - Sales in Japan (in millions)

In [40]:
data_new[['NA_Sales','EU_Sales','JP_Sales']].corr()

NA_Sales  EU_Sales  JP_Sales
NA_Sales  1.000000  0.912718  0.270399
EU_Sales  0.912718  1.000000  0.276157
JP_Sales  0.270399  0.276157  1.000000

Итак, корреляция между продажами в Северной Америке и Европой 0.911, корреляция между Северной Америкой и Японией 0.270, а корреляция между продажами в Японии и Европе 0.276. Самая высокая корреляция между продажами в Северной Америке и Европе 0.912, это означает, что есть связь между продажами в этих двух континентах. Корреляции между продажами в Японии и другими континентами низкая, это ознаает, что связи между продажами почти нет.

Самая большая связь между Северной Америкой и Европой, а самая низкая связь между Северной Америкой и Японией.